index.js

```diff
@@ -27,17 +27,24 @@ function validateResponse (fastify, opts, next) {
   function buildHook (schema) {
     const statusCodes = {}
     for (const statusCode in schema) {
-      statusCodes[statusCode] = ajv.compile(schema[statusCode])
+      const responseSchema = schema[statusCode]
+      statusCodes[statusCode] = ajv.compile(
+        // replicate getSchemaAnyway from fastify
+        // https://github.com/fastify/fastify/blob/23371945d01c270af24f4a5b7e2e31c4e806e6b3/lib/schemas.js#L113
+        !responseSchema.type && !responseSchema.properties
```

Comment:
Can you copy the whole function? `oneOf`, `anyOf`, `allOf` ... are supported in the original function.

Reply:
Do you mean `ajv.compile` function?

```js
statusCodes[statusCode] = !responseSchema.type && !responseSchema.properties ? ajv.compile(...) : ajv.compile(...);
```

Or do you mean entire buildHook? I haven't also understood about `oneOf` and other.

Reply:
Yes. the `ajv.compile` function.
You are using a condition statement for the `schema` which do not support `oneOf`, `anyOf`, etc.

Reply:
Ah, I see. I've skipped this part of logic:

![image](https://user-images.githubusercontent.com/2536916/126873028-80d9541b-7e4d-4c32-b4dd-d18fefde89d6.png)

Reply:
I've copied an entire function from `fastify`